plugins/qsi-dock/fu-qsi-dock-mcu-device.c

```diff
@@ -271,7 +271,7 @@ fu_qsi_dock_mcu_device_write_chunk(FuQsiDockMcuDevice *self,
 	g_autoptr(GPtrArray) chunks = NULL;
 
 	chunks = fu_chunk_array_new(fu_chunk_get_data(chk_page),
-				    fu_chunk_get_data_sz(chk_page),
+				    FU_QSI_DOCK_EXTERN_FLASH_PAGE_SIZE,
```

Comment:
I don' understand this change; won't this be exactly the same value? If it's not exactly the same value then we're over-reading the buffer which is going to make valgrind and asan explode.

Is the issue perhaps that fu_firmware_get_bytes() isn't aligned to FU_QSI_DOCK_EXTERN_FLASH_PAGE_SIZE? In which case we need to use fu_bytes_align() rather than just overread a C buffer which probably has junk.

Reply:
I've switched this to use `fu_bytes_align()` instead.